### PR TITLE
Fix permissions for access permissions test

### PR DESCRIPTION
**Pull request by nadjaheitmann**
Fix test:
`bundle exec rake test TEST="test/unit/foreman/access_permissions_test.rb"`

### CODE REVIEW

File: lib/foreman_scc_manager/engine.rb [hunk 1]
```diff
@@ -60,15 +60,32 @@ class Engine < ::Rails::Engine
...
         # Add a new role called 'SccManager' if it doesn't exist
         role 'SccManager',
-             %i[view_scc_accounts use_scc_accounts new_scc_accounts edit_scc_accounts delete_scc_accounts sync_scc_accounts],
+             %i[view_scc_accounts use_scc_accounts new_scc_accounts edit_scc_accounts \
+                delete_scc_accounts sync_scc_accounts test_connection_scc_accounts \
+                view_scc_products subscribe_scc_products],
              'Role granting permissons to manage SUSE Subscriptions'
 
         role 'SccViewer',
-             %i[view_scc_accounts use_scc_accounts sync_scc_accounts create_products view_products],
```
Comment (m-bucher) [inline lib/foreman_scc_manager/engine.rb:71]: I think we need `create_products` to Create the Katello product.

File: db/migrate/20210713092440_add_permissions.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  PERMISSION_NAMES_ALL = [
+    'test_connection_scc_accounts',
+    'view_scc_products',
+    'subscribe_scc_products'
+  ].freeze
+
+  def up
+    Permission.create!(:name => 'test_connection_scc_accounts', :resource_type => 'SccAccount')
+    PRODUCT_PERMISSION_NAMES.each do |p|
+      Permission.create!(:name => p, :resource_type => SccProduct)
```
Comment (m-bucher) [inline db/migrate/20210713092440_add_permissions.rb:15]: You could avoid having two arrays with almost the same content by doing something like the following:
```suggestion
  def up
    permissions = PERMISSION_NAMES_ALL.clone
    Permission.create!(:name => permissions.pop, :resource_type => 'SccAccount')
    permissions.each do |p|
      Permission.create!(:name => p, :resource_type => SccProduct)
```

**Review by m-bucher — APPROVED** [review]

File: db/migrate/20210713092440_add_permissions.rb (hunk 2 above)
Comment (nadjaheitmann) [reply db/migrate/20210713092440_add_permissions.rb:15]: I agree that this is much better code, but is it more readable?

File: foreman_scc_manager.gemspec [hunk 3]
```diff
@@ -16,7 +16,7 @@ Gem::Specification.new do |s|
   s.files = Dir['{app,config,db,lib,locale}/**/*'] + ['LICENSE', 'Rakefile', 'README.md']
   s.test_files = Dir['test/**/*']
 
-  s.add_development_dependency 'rdoc', '~> 4'
+  s.add_development_dependency 'rdoc', '>= 6.3.1'
```
[the remark this answers is not in the document]
Comment (nadjaheitmann) [reply foreman_scc_manager.gemspec]: Fixed it ;)

File: db/migrate/20210713092440_add_permissions.rb (hunk 2 above)
Comment (m-bucher) [reply db/migrate/20210713092440_add_permissions.rb:15]: I agree that messing with a copy of the array is not that nice.

Alternatively, I'd settle for always using `PRODUCT_PERMISSION_NAMES` and doing the destroy of `test_connection_scc_accounts` separately or adding it to the array locally:
```ruby
  def down
    (PRODUCT_PERMISSION_NAMES + ['test_connection_scc_accounts']).each do |p|
      Permission.where(:name => p).destroy_all
    end
  end

Comment (nadjaheitmann) [reply db/migrate/20210713092440_add_permissions.rb:15]: I like that one!

**Pull request merged**